FreeSurfer/LongitudinalFreeSurferPipeline.sh

```diff
@@ -0,0 +1,418 @@
+LongDIR="${StudyFolder}/${SubjectID}.long.${TemplateID}/T1w"
+mkdir -p "${LongDIR}"
+for Session in ${Sessions} ; do
+  Source="${StudyFolder}/${Session}/T1w/${Session}"
+  Target="${LongDIR}/${Session}"
+  log_Msg "Creating a link: ${Source} => ${Target}"
+  ln -sf ${Source} ${Target}
+done
+
+
+if (( generate_timepoints_only != 1 )); then 
+
+	# ----------------------------------------------------------------------
+	log_Msg "Creating the base template: ${TemplateID}"
+	# ----------------------------------------------------------------------
+	# backup template dir if it exists
+	if [ -d "${LongDIR}/${TemplateID}" ]; then
+	  TimeStamp=`date +%Y-%m-%d_%H.%M.%S.%6N`
+	  log_Msg "Base template dir: ${LongDIR}/${TemplateID} already exists, backing up to ${LongDIR}/${TemplateID}.${TimeStamp}"
+	  mv ${LongDIR}/${TemplateID} ${LongDIR}/${TemplateID}.${TimeStamp}
+	fi
+
+	recon_all_cmd="recon-all.v6.hires"
+	recon_all_cmd+=" -sd ${LongDIR}"
+	recon_all_cmd+=" -base ${TemplateID}"
+	for Session in ${Sessions} ; do
+	  recon_all_cmd+=" -tp ${Session}"
+	done
+	recon_all_cmd+=" -all"
+
+	if [ ! -z "${recon_all_seed}" ]; then
+	  recon_all_cmd+=" -norandomness -rng-seed ${recon_all_seed}"
+	fi
+
+
+	#---------------------------------------------------------------------------------------
+	log_Msg "Running the recon-all to generate common template"
+	#---------------------------------------------------------------------------------------
+
+	#recon_all_cmd+=(${extra_reconall_args_base[@]+"${extra_reconall_args_base[@]}"})
+	recon_all_cmd+=" ${extra_reconall_args_base[@]}"
+	echo "recon_all_cmd:"
+	echo ${recon_all_cmd}
+	log_Msg "...recon_all_cmd: ${recon_all_cmd}"
+
+	${recon_all_cmd}
+	return_code=$?
+	if [ "${return_code}" != "0" ]; then
+	  log_Err_Abort "recon-all command failed with return_code: ${return_code}"
+	fi
+fi
+if (( generate_template_only != 1)); then 
+
+	# ----------------------------------------------------------------------
+	log_Msg "Running the longitudinal recon-all on each timepoint"
+	# ----------------------------------------------------------------------
+
+	for Session in ${Sessions} ; do
+	  log_Msg "Running longitudinal recon all for session: ${Session}"
+	  recon_all_cmd="recon-all.v6.hires"
+	  recon_all_cmd+=" -sd ${LongDIR}"
+	  recon_all_cmd+=" -long ${Session} ${TemplateID} -all"
+	  
+	  #recon_all_cmd+=(${extra_reconall_args_long[@]+"${extra_reconall_args_long[@]}"})
+	  recon_all_cmd+=" ${extra_reconall_args_long[@]}"
+	  T2w=${StudyFolder}/${Session}/T1w/T2w_acpc_dc_restore.nii.gz
+	  
+	  if [ -f "$T2w" ]; then 
+		  recon_all_cmd+=" -T2 $T2w"
+	  else
+	  	  log_Msg "WARNING: No T2-weighed image $T2w, T2-weighted processing will not run."
+	  fi
+	  	  
+	  T1w=${StudyFolder}/${Session}/T1w/T1w_acpc_dc_restore.nii.gz
+	  emregmask=${StudyFolder}/${Session}/T1w/T1w_acpc_dc_restore_brain.nii.gz
+	  
+	  if [ -f "$T1w" -a -f "$emregmask" ]; then 
+	  	recon_all_cmd+=" -emregmask $emregmask" #-i $T1w 
```

Comment:
The logic here seems to be "emregmask must exist, and will always be added, otherwise error, and if the T1w doesn't exist, also error".  If this is correct, the addition of the `-emregmask` argument doesn't need to be in a conditional at all.

Reply:
We always use an -emregmask for FreeSurfer, so it would be an error.

Reply:
$T1w is not an actual input to FS-long, so I'm unclear why we are even defining that variable and checking for its existence here.

Reply:
Ok, addressed.

Reply:
Please push the commit that resolves it before clicking "resolve conversation".